pytorch_lightning/trainer/connectors/profiler_connector.py

```diff
@@ -11,16 +11,40 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License
-from pytorch_lightning.profiler import PassThroughProfiler, SimpleProfiler
+
+from typing import Union
+
+from pytorch_lightning.profiler import BaseProfiler, PassThroughProfiler, SimpleProfiler, AdvancedProfiler
+from pytorch_lightning.utilities import rank_zero_warn
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
 
 
 class ProfilerConnector:
 
     def __init__(self, trainer):
         self.trainer = trainer
 
-    def on_trainer_init(self, profiler):
-        # configure profiler
-        if profiler is True:
-            profiler = SimpleProfiler()
+    def on_trainer_init(self, profiler: Union[BaseProfiler, bool, str]):
+
+        if profiler and not isinstance(profiler, (bool, str, BaseProfiler)):
+            # TODO: Update exception on deprecation of bool
```

Comment:
You already deprecated bool, I think you meant:

```suggestion
            # TODO: Update exception on removal of bool
```